cfme/utils/appliance/implementations/ui.py

```diff
@@ -287,6 +276,20 @@ def check_for_badness(self, fn, _tries, nav_args, *args, **kwargs):
             self.go(_tries, *args, **go_kwargs)
             # If there is a rails error past this point, something is really awful
 
+    def check_for_badness(self, fn, _tries, nav_args, *args, **kwargs):
+        if getattr(fn, '_can_skip_badness_test', False):
```

Comment:
**QUESTION:** What is fn?

Reply:
fn is 'function', check_for_badness is a wrapper for step/am_i_here and other NavigateStep methods.

Reply:
It's all a little tiny, insignificant;y small bit of a mess :)